rpcclient/rawtransactions.go

```diff
@@ -233,9 +233,15 @@ func (r FutureCreateRawTransactionResult) Receive() (*wire.MsgTx, error) {
 
 	// Deserialize the transaction and return it.
 	var msgTx wire.MsgTx
-	if err := msgTx.Deserialize(bytes.NewReader(serializedTx)); err != nil {
-		return nil, err
+	// we try both the new and old encoding format
+	witnessErr := msgTx.Deserialize(bytes.NewReader(serializedTx))
+	if witnessErr != nil {
+		legacyErr := msgTx.DeserializeNoWitness(bytes.NewReader(serializedTx))
+		if legacyErr != nil {
+			return nil, legacyErr
+		}
```

Comment:
If both return an error, would the call to `msgTx.Deserialize(...)` return something useful that isn't already covered by `msgTx.DeserializeNoWitness(...)`?

Reply:
Sorry, I'm not sure I understand what you mean. Is this about which error to return?

Reply:
Sort of, yeah. Although I also just looked at the code and it's fine to do it this way. Let's say both `msgTx.Deserialize` and `msgTx.DeserializeNoWitness` both return an error. In this case, we go down the `if witnessErr != nil` branch and eventually return `(nil, legacyErr)`. But there is a `witnessErr` which is non-nil and I was wondering if it might make sense to return both error messages.
I just checked the code though and both just make calls to `msg.BtcDecode` so it should be fine.